purpleair-aqi.js

```diff
@@ -89,184 +128,191 @@ const levelAttributes = [
   },
 ];
 
-// Get level attributes for AQI
-function getLevelAttributes(level, attributes) {
-  let applicableAttributes = attributes
-    .filter((c) => level > c.threshold)
-    .sort((a, b) => b.threshold - a.threshold);
-  return applicableAttributes[0];
+/**
+ * Get the EPA adjusted PPM
+ *
+ * @param {SensorData} sensorData
+ * @returns {number} EPA draft adjustment for wood smoke and PurpleAir from https://cfpub.epa.gov/si/si_public_record_report.cfm?dirEntryId=349513&Lab=CEMM&simplesearch=0&showcriteria=2&sortby=pubDate&timstype=&datebeginpublishedpresented=08/25/2018
+ */
+function computePM(sensorData) {
+  const adj1 = Number.parseInt(sensorData.adj1, 10);
+  const adj2 = Number.parseInt(sensorData.adj2, 10);
+  const hum = Number.parseInt(sensorData.hum, 10);
+  const dataAverage = (adj1 + adj2) / 2;
+
+  return 0.52 * dataAverage - 0.085 * hum + 5.71;
 }
 
-// Function to get the EPA adjusted PPM
-function computePM(data) {
-  let adj1 = parseInt(data.adj1, 10);
-  let adj2 = parseInt(data.adj2, 10);
-  let hum = parseInt(data.hum, 10);
-  let dataAverage = ((adj1 + adj2)/2);
-
-  // Apply EPA draft adjustment for wood smoke and PurpleAir
-  // from https://cfpub.epa.gov/si/si_public_record_report.cfm?dirEntryId=349513&Lab=CEMM&simplesearch=0&showcriteria=2&sortby=pubDate&timstype=&datebeginpublishedpresented=08/25/2018
-
-  return ((0.52 * dataAverage) - (.085 * hum) + 5.71);
-}
-
-// Function to get AQI number from PPM reading
+/**
+ * Get AQI number from PPM reading
+ *
+ * @param {number} pm
+ * @returns {number|'-'}
+ */
 function aqiFromPM(pm) {
-  if (pm > 350.5) {
-    return calcAQI(pm, 500.0, 401.0, 500.0, 350.5);
-  } else if (pm > 250.5) {
-    return calcAQI(pm, 400.0, 301.0, 350.4, 250.5);
-  } else if (pm > 150.5) {
-    return calcAQI(pm, 300.0, 201.0, 250.4, 150.5);
-  } else if (pm > 55.5) {
-    return calcAQI(pm, 200.0, 151.0, 150.4, 55.5);
-  } else if (pm > 35.5) {
-    return calcAQI(pm, 150.0, 101.0, 55.4, 35.5);
-  } else if (pm > 12.1) {
-    return calcAQI(pm, 100.0, 51.0, 35.4, 12.1);
-  } else if (pm >= 0.0) {
-    return calcAQI(pm, 50.0, 0.0, 12.0, 0.0);
-  } else {
-    return "-";
-  }
+  if (pm > 350.5) return calculateAQI(pm, 500.0, 401.0, 500.0, 350.5);
+  if (pm > 250.5) return calculateAQI(pm, 400.0, 301.0, 350.4, 250.5);
+  if (pm > 150.5) return calculateAQI(pm, 300.0, 201.0, 250.4, 150.5);
+  if (pm > 55.5) return calculateAQI(pm, 200.0, 151.0, 150.4, 55.5);
+  if (pm > 35.5) return calculateAQI(pm, 150.0, 101.0, 55.4, 35.5);
+  if (pm > 12.1) return calculateAQI(pm, 100.0, 51.0, 35.4, 12.1);
+  if (pm >= 0.0) return calculateAQI(pm, 50.0, 0.0, 12.0, 0.0);
+  return "-";
```

Comment:
Perhaps an error should be thrown here, since if `pm` is less than `0` then something has gone wrong.

Something like:
```suggestion
  throw new Error("pm should not be less than zero")
```